src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -258,10 +260,21 @@ private void addOptions(Template template, Options options, Infrastructure infra
             if (ports != null) {
                 securityGroupName += "-" + UUID.randomUUID();
                 String sgDescription = "Auto generated security group to authorize the ports " + Arrays.toString(ports);
-                AWSSubnetApi awsSubnetApi = getAWSSubnetApi(infrastructure);
-                String vpcId = awsSubnetApi.describeSubnetsInRegion(region, options.getSubnetId()).get(0).getVpcId();
                 CreateSecurityGroupOptions sgOptions = new CreateSecurityGroupOptions();
-                sgOptions.vpcId(vpcId);
+                String subnetId = options.getSubnetId();
```

Comment:
If the front-end allows empty values for the SubnetId, it would be better to use the method `isNullOrEmpty(String string)` of google.common (instead of `!= null`)

Reply:
If it is empty I get a `null` in any case in the options so we do not have any difference. But I will change it anyway as you suggest.